pyamplipi/models.py

```diff
@@ -29,8 +29,8 @@ class Source(BaseModel):
 
 class SourceUpdate(BaseModel):
     """ Partial reconfiguration of an audio Source """
-    name: Optional[str]
-    input: Optional[str]  # 'None', 'local', 'stream=ID'
+    name: Optional[str] = None
+    input: Optional[str] = None  # 'None', 'local', 'stream=ID'
```

Comment:
We create `SourceUpdate`s in places, and don't pass these args in. However, just because we marked them `Optional` does not mean they get a default value of `None`. There's a lot of these we could fixup here, but I'm hesitant to poke vast swaths of code (and the spots I did poke mean our project passes mypy for now.) Let me know if you feel like I should hit up every `Optional` with this so downstream projects will also typecheck well without having to put in fussy nonetypes everywhere, and I will.

Reply:
re: "validat[ing] it doesn't break our HA plugin", [`pyamplipi` is version pinned](https://github.com/micro-nova/hacs_amplipi/blob/073b97d212e685629902ce3f68b937f6c1db5d29/custom_components/amplipi/manifest.json#L6), and while this might break the API contract (do we even do semantic versioning here?) it won't break the plugin for existing or new installs.